docs/specs/app-initiated-sso-to-web.md

```diff
@@ -0,0 +1,182 @@
+# App-initiated SSO to Web
+
+- [Abstract](#abstract)
+- [The Flow](#the-flow)
+- [Native App Authorization Extensions](#native-app-authorization-extensions)
+- [Token Exchange Profile for App Initiated SSO to Web](#token-exchange-profile-for-app-initiated-sso-to-web)
+  - [Defined token type identifiers](#defined-token-type-identifiers)
+  - [Token Exchange Request](#token-exchange-request)
+  - [Token Exchange Response](#token-exchange-response)
+- [Exchanging for browser cookie of a session with the authorization endpoint](#exchanging-for-browser-cookie-of-a-session-with-the-authorization-endpoint)
+- [Security Considerations](#security-considerations)
+  - [Binding Tokens To Device](#binding-tokens-to-device)
+- [Changes In SDKs](#changes-in-sdks)
+- [Usecases](#usecases)
+
+## Abstract
+
+This document describes a mechanism that allows a cookie based web app to share the authentication obtained by a mobile app.
+
+This document is drafted based on [OpenID Connect Native SSO for Mobile Apps 1.0](https://openid.net/specs/openid-connect-native-sso-1_0.html).
+
+## The Flow
+
+```mermaid
+sequenceDiagram
+    participant app1 as Mobile App #35;1
+    participant Web as Web App #35;2 <br/>(Cookie Based)<br/>https://www.example.com
+    participant Authgear as Authgear <br/> https://auth.example.com
+    app1->>+Authgear: [1]GET https://auth.example.com/oauth2/authorize<br/>client_id=1<br/>scope=x_device_browser_session_token<br/>redirect_uri=app://redirect
+    Note right of Authgear: Alice enter user credentials
+    Authgear->>-app1: [2]Redirect to app://redirect?code=CODE
+    app1->>+Authgear: [3]POST https://auth.example.com/oauth2/token<br/>code=CODE
+    Authgear->>-app1: [4]Response with<br/>access_token=ACCESS_TOKEN<br/>refresh_token=REFRESH_TOKEN<br/>id_token=ID_TOKEN<br/>device_secret=DEVICE_SECRET
+    app1->>+Authgear: [5]POST https://auth.example.com/oauth2/token<br/>grant_type=token-exchange<br/>client_id=2<br/>subject_token_type=urn:ietf:params:oauth:token-type:id_token<br/>subject_token=ID_TOKEN<br/>actor_token_type=urn:x-oath:params:oauth:token-type:device-secret<br/>actor_token=DEVICE_SECRET<br/>requested_token_type=urn:x-oath:params:oauth:token-type:device-browser-session-token
```

Comment:
- scope: `https://authgear.com/scopes/app-initiated-sso-to-web`
- token identifier: `...:app-initiated-sso-to-web-token`
- parameter name in authorization endpoint: `x_app_initiated_sso_to_web_token`